tests/functional/sign_in_recovery_code.js

```diff
@@ -108,5 +109,52 @@ registerSuite('recovery code', {
         // about:accounts will take over post-verification, no transition
         .then(testIsBrowserNotified('fxaccounts:login'));
     },
+
+    'can regenerate recovery code when low': function () {
+      return this.remote
+        .then(click(selectors.SIGNIN_RECOVERY_CODE.DONE_BUTTON))
+        .then(click(selectors.SETTINGS.SIGNOUT))
+        .then(openPage(SYNC_SIGNIN_URL, selectors.SIGNIN.HEADER, {
+          query: {}, webChannelResponses: {
+            'fxaccounts:can_link_account': {ok: true},
+            'fxaccounts:fxa_status': {capabilities: null, signedInUser: null},
+          }
+        }))
+
+        .then(fillOutSignIn(email, PASSWORD))
+        .then(testElementExists(selectors.TOTP_SIGNIN.HEADER))
+        .then(click(selectors.SIGNIN_RECOVERY_CODE.LINK))
+
+        .then(type(selectors.SIGNIN_RECOVERY_CODE.INPUT, recoveryCode))
+        .then(click(selectors.SIGNIN_RECOVERY_CODE.SUBMIT))
+
+        .then(testIsBrowserNotified('fxaccounts:login'))
```

Comment:
If the browser should only be notified after the user fills out the signin recovery input, a `noSuchBrowserNotification` before the submit might be useful.